middlewares/sponsor.middleware.js

```diff
@@ -45,7 +53,50 @@ function parseSponsor (req, res, next) {
     next();
 }
 
+/**
+ * Verifies that account is confirmed and of proper type from the account ID passed in req.body.accountId
+ * @param {{body: {accountId: ObjectId}}} req 
+ * @param {*} res 
+ * @param {(err?) => void} next 
+ */
+async function validateConfirmedStatus(req, res, next) {
+    const account = await Services.Account.findById(req.body.accountId);
+
+    if (account && account.confirmed && account.accountType === Constants.SPONSOR) {
```

Comment:
Consider splitting logic here so that the error code / message is appropriate depending on what validation failed.

Reply:
Updated